src/processor_arm.cpp

```diff
@@ -721,7 +721,7 @@ static NOINLINE std::pair<uint32_t,FeatureList<feature_sz>> _get_host_cpu()
     else if (cpu_name.find("M2") != StringRef ::npos)
         return std::make_pair((uint32_t)CPU::apple_m2, Feature::apple_m2);
     else if (cpu_name.find("M3") != StringRef ::npos)
-        return std::make_pair((uint32_t)CPU::apple_m3, Feature::apple_m3);
+        return std::make_pair((uint32_t)CPU::apple_m2, Feature::apple_m2); // m3 doesn't exist in LLVM 16 for 1.11
```

Comment:
I don't understand -- aren't `CPU::apple_m3` and `Feature::apple_m3` constants defined in this very same file?  And in fact `Feature::apple_m2` and `Feature::apple_m4` seem to have identical definition:
```
constexpr auto apple_m2 = armv8_5a_crypto | get_feature_masks(dotprod,fp16fml, fullfp16, sha3, i8mm, bf16);
constexpr auto apple_m3 = armv8_5a_crypto | get_feature_masks(dotprod,fp16fml, fullfp16, sha3, i8mm, bf16);
```

Reply:
Yes. But LLVM uses the name itself in Enzyme, and it doesn't recognize the name apple-m3. We could also add m4 handling but we'd need to know what features it has.

Reply:
The reference to `Feature::apple_m4` here was a typo, I meant `Feature::apple_m3`.

What baffled me was that in the resulting compiled binary, `Feature::apple_m3` and `Feature::apple_m2` should be indistinguishable (as they have identical definition).

But I see now that the problem likely is that `CPU::apple_m2` and `CPU::apple_m3` are two different integers, which are passed to e.g. `find_cpu` and `find_cpu_name`, and of course for that it is then plausible that it may affect other packages.

Sorry for the noise :-)

Reply:
Don't worry. This code is a mess :)